src/dissect/dissect.c

```diff
@@ -860,6 +860,9 @@ static int action_dissect(DissectedImage *m, LoopDevice *d) {
                 return log_error_errno(r, "Failed to acquire image metadata: %m");
         else if (arg_json_format_flags & JSON_FORMAT_OFF) {
 
+                if (m->image_name && !streq_ptr(m->image_name, bn))
```

Comment:
bn is always valid, and m->image_name is explicitly checked. So _ptr is not needed?